app/services/census_authorization_handler.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+# Checks the authorization against the census for Barcelona.
+class CensusAuthorizationHandler < Decidim::AuthorizationHandler
+  include ActionView::Helpers::SanitizeHelper
+
+  attribute :document_number, String
+  attribute :document_type, String
+  attribute :postal_code, String
+  attribute :date_of_birth, Date
+
+  validates :document_number, :document_type, :postal_code, :date_of_birth,
+            presence: true
+  validates :document_type, inclusion: { in: %w(dni nie passport) }
+
+  # The only method that needs to be implemented for an authorization handler.
+  # Here you can add your business logic to check if the authorization should
+  # be created or not, you should return a Boolean value.
+  def authorized?
+    valid? && response.xpath("//codiRetorn").text == "01"
```

Comment:
Please reimplement this as a validation (`//codiRetorn`). I case the return code is invalid, add an error to `document_number`.